### DISCUSSION TITLE
Tanka GitOps operator?

### DISCUSSION THREAD

**Discussion by gitbluf** [grafana/tanka#740] [General]
Hey guys,

Like the project, but I was thinking if there are any plans about developing Tanka operator(for example as an extension to flux ecosystem)?

Currently the only way I can see this happening is having a CI that exports manifests and commits them(then flux takes over).

Let me know what you think

**Comment by Duologic** [comment]
At Grafana, we do the latter, render yaml in CI and let Flux handle the rest.

**Reply by Almenon** [reply]
I'm curious how this works step by step (if you're able to share). For example, do you commit the yaml back to the git repo?

**Reply by Duologic** [reply]
In general it's `tk export` and commit that to git, every commit to master on the source has a commit on the master of the repo containing the export. CI gets serialized to prevent race conditions. PRs create branches on the export repo and the CI job comments a link to diff the output, this gives a very useful feedback loop.

**Comment by Almenon** [comment]
Thanks! Why not save the export in a folder in the same repo rather than a
different one?

**Reply by Duologic** [reply]
The repo can only be written to by automation, which prevents (accidental) tampering, the source repo has lots of traffic too, so would cause conflicts all the time.
